k8s/shipwright/unsecured/v1beta1/clusterbuildstrategy.yml

```diff
@@ -0,0 +1,294 @@
+      defaults: []
+    - name: PLATFORM_DIR
+      description: The name of the platform directory.
+      default: empty-dir
+    - # Update the user id to use new paketo builder image till we can use: https://github.com/shipwright-io/community/blob/main/ships/0036-runAs-for-supporting-steps.md
+      name: CNB_USER_ID
+      description: The user ID of the builder image user.
+      default: "1002"
+    - name: CNB_GROUP_ID
+      description: The group ID of the builder image user.
+      default: "1000"
+    - name: USER_HOME
+      description: Absolute path to the user's home directory.
+      default: /tekton/home
+  steps:
+    - name: prepare
+      image: docker.io/library/bash:5.1.4
+      args:
+        - -c
+        - |
+          set -e
+
+          # TODO: To be reviewed
+          echo "> Creating the cache directory if it is not empty"
+          if [ ! -d "$DIRECTORY" ]; then
+            echo "> Creating cache: /layers/$(params.CACHE_DIR_NAME)"
+            mkdir -p /layers/$(params.CACHE_DIR_NAME)
+            chown -R "$(params.CNB_USER_ID):$(params.CNB_GROUP_ID)" /layers/$(params.CACHE_DIR_NAME)
+          fi
+          
+          # TODO: To be reviewed as shipwright don't support like Tekton workspaces
+          #if [[ "$(workspaces.cache.bound)" == "true" ]]; then
+          #  echo "> Setting permissions on '$(workspaces.cache.path)'..."
+          #  chown -R "$(params.CNB_USER_ID):$(params.CNB_GROUP_ID)" "$(workspaces.cache.path)"
+          #fi
+
+          for path in "/tekton/home" "/layers" "$(workspaces.source.path)"; do
+            echo "> Setting permissions on '$path'..."
+            chown -R "$(params.CNB_USER_ID):$(params.CNB_GROUP_ID)" "$path"
+          done
+
+          echo "> Parsing additional configuration..."
+          parsing_flag=""
+          envs=()
+          for arg in "$@"; do
+              if [[ "$arg" == "--env-vars" ]]; then
+                  echo "-> Parsing env variables..."
+                  parsing_flag="env-vars"
+              elif [[ "$parsing_flag" == "env-vars" ]]; then
+                  envs+=("$arg")
+              fi
+          done
+
+          echo "> Processing any environment variables..."
+          ENV_DIR="/platform/env"
+
+          echo "--> Creating 'env' directory: $ENV_DIR"
+          mkdir -p "$ENV_DIR"
+
+          for env in "${envs[@]}"; do
+              IFS='=' read -r key value string <<< "$env"
+              if [[ "$key" != "" && "$value" != "" ]]; then
+                  path="${ENV_DIR}/${key}"
+                  echo "--> Writing ${path}..."
+                  echo -n "$value" > "$path"
+              fi
+          done
+        # That's the separator between the shell script and its args
+        - --
+        - --env-vars
+        - $(params.ENV_VARS[*])
+      volumeMounts:
+        - name: layers-dir
+          mountPath: /layers
+        - name: empty-dir #name: --> DON'T WORK : $(params.PLATFORM_DIR)
```

Comment:
`name` must reference a named `volume` above. I'd like to know more about the `PLATFORM_DIR` parameter - is this something end users/app developers can change?

If each of the volumes above is intended to hold cached data, I recommend only having one volume based on my recent experiments. You can use the `subPath` field to specify a directory within said volume.  And you can mount the same `volume` into a container multiple times, as long as each goes to a unique `mountPath`.

If that last sentence sounds like a symbolic link to you, _you are not wrong_!

Reply:
> I'd like to know more about the `PLATFORM_DIR` parameter - is this something end users/app developers can change?

Yes. This is a CNB env var that users can change if needed. Otherwise, they will accept the default value. See the buildpacks spec: https://github.com/buildpacks/spec/blob/main/platform.md#inputs-1

Reply:
> `name` must reference a named `volume` above

I hard coded the name when I experimented months ago shipwright as passing the name of the volume/workspace wazs not working. This is the reason why you see within the YAML file `name: empty-dir #name: --> DON'T WORK : $(params.PLATFORM_DIR) ...`